packages/components/icon/src/icon.js

```diff
@@ -39,12 +41,69 @@ export class TSIcon extends TSElement {
 			/** 90, 180, 270 */
 			rotate: { type: Number, reflect: true },
 			/** 'h', 'horizontal', 'v', 'vertical' */
-			flip: { type: String, reflect: true }
+			flip: { type: String, reflect: true },
+			/** INTERNAL svg content to be rendered inside icon */
+			svgContent: { type: String, reflect: false, attribute: null }
 		};
 	}
 
+	attributeChangedCallback(name, oldVal, newVal) {
+		super.attributeChangedCallback(name, oldVal, newVal);
+		switch (name) {
+			case 'icon':
+				if (!this.src && newVal !== null) {
+					window.customElements.whenDefined('ts-icon').then(() => {
+						this.getIcon();
+					});
+				}
+				break;
+			case 'src':
+				if (newVal !== null) {
+					window.customElements.whenDefined('ts-icon').then(() => {
+						this.getSrc();
+					});
+				}
+				break;
+			// do nothing
+		}
+	}
+
+	async getIcon() {
+		if (!this.iconCache.has(this.icon) && this.iconCache.get(this.icon) !== '') {
```

Comment:
I don't know. A cache value should never be an empty string. Maybe disallow adding an empty string, or otherwise clean up/validate the cache and remove items with empty string values.